instantiate from loader gives a type error

I tried to `instantiate` with asc loader, lilke this:
```
instantiate<{}>(wasm, {
  env: {
    abort: (msg: number, file: number, line: number, column: number) => {
      console.error("abort called at" + file + " " + line + ":" + column);
    }
  },
})
```

And It gives a following error:

```
error TS2411: Property 'env' of type '{ memory?: Memory | undefined; table?: Table | undefined; seed?: (() => number) | undefined; abort?(msg: number, file: number, line: number, column: number): void; trace?(msg: number, numArgs?: number | undefined, ...args: number[]): void; } | undefined' is not assignable to string index type 'Record<string, ImportValue>'.
```

I'm using `assemblyscript@0.9.4-nightly.20200325`.

The error was gone when I tried this:

```ts
export interface WasmEnv {
  env?: {
    memory?: WebAssembly.Memory,
    table?: WebAssembly.Table,
    seed?: () => number,
    abort?(msg: number, file: number, line: number, column: number): void,
    trace?(msg: number, numArgs?: number, ...args: number[]): void
  };
}

/** WebAssembly imports with two levels of nesting. */
export type Imports = WasmEnv & Record<string, Record<string, ImportValue>>;
```

instead of the existing type:
```ts
/** WebAssembly imports with two levels of nesting. */
export interface Imports extends Record<string, Record<string, ImportValue>> {
  env?: {
    memory?: WebAssembly.Memory,
    table?: WebAssembly.Table,
    seed?: () => number,
    abort?(msg: number, file: number, line: number, column: number): void,
    trace?(msg: number, numArgs?: number, ...args: number[]): void
  };
}
```

Yeah, so maybe the type definitions of the loader needs to be fixed.

--- a/lib/loader/index.d.ts
+++ b/lib/loader/index.d.ts
@@ -8,15 +8,15 @@ export interface ResultObject {
 type ImportValue = Function | WebAssembly.Global | WebAssembly.Memory | WebAssembly.Table | number;
 
 /** WebAssembly imports with two levels of nesting. */
-export interface Imports extends Record<string, Record<string, ImportValue>> {
+export type Imports = {
   env?: {
-    memory?: WebAssembly.Memory,
-    table?: WebAssembly.Table,
-    seed?: () => number,
-    abort?(msg: number, file: number, line: number, column: number): void,
-    trace?(msg: number, numArgs?: number, ...args: number[]): void
-  };
-}
+    memory?: WebAssembly.Memory;
+    table?: WebAssembly.Table;
+    seed?: () => number;
+    abort?(msg: number, file: number, line: number, column: number): void;
+    trace?(msg: number, numArgs?: number, ...args: number[]): void;
+  } & Record<string, ImportValue>;
+} & Record<string, Record<string, ImportValue>>;
 
 /** Utility mixed in by the loader. */
 export interface ASUtil {

--- a/lib/loader/index.d.ts
+++ b/lib/loader/index.d.ts
@@ -12,7 +12,7 @@ export type Imports = {
   env?: {
     memory?: WebAssembly.Memory;
     table?: WebAssembly.Table;
-    seed?: () => number;
+    seed?(): number;
     abort?(msg: number, file: number, line: number, column: number): void;
     trace?(msg: number, numArgs?: number, ...args: number[]): void;
   } & Record<string, ImportValue>;

